### PR TITLE
Fix/report naming

### PR DESCRIPTION
**Pull request by Donnype**
### Changes

- Adds the right naming logic in the report runner.
- Introduces a first integration test for the report runner.
- Some typing fixes.

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3632

### Demo

See the template and outputted names in the integration test: this is what we expect now (@Rieven right?).

### QA notes

Please go through the generate report flow and verify that the names appear as expected. Also see the issue for what we want to avoid.

---

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.
- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/views/mixins.py [hunk 1]
```diff
@@ -268,6 +276,9 @@ def save_report(self, report_names: list) -> Report:
         # Save the child reports to bytes
         for ooi, types in report_data.items():
             for report_type, data in types.items():
-                save_report_raw(bytes_client, data=data | self.get_input_data())
+                bytes_client.upload_raw(
```
Comment (Rieven) [inline rocky/reports/views/mixins.py:279]: This needs some changes in base.py, get_input_data has "input_data"
image

Comment (Rieven) [reply rocky/reports/views/mixins.py:279]: I will fix this

File: rocky/tests/integration/test_report_runner.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    reports = octopoes_api_connector.list_reports(valid_time)
+    assert reports.count == 1
+
+    report, subreports = reports.items[0]
+    assert len(subreports) == 2
+
+    assert report.name == "Concatenated report for 2 objects"
+    assert {x.name for x in subreports} == {
+        "DNS Report for Hostname|test|a.example.com in 2024",
+        "DNS Report for Hostname|test|example.com in 2024",
```
Comment (Rieven) [inline rocky/tests/integration/test_report_runner.py:71]: Yes, this is a possible scenario where the user enters: {report_type} for {ooi} in %Y

File: rocky/reports/runner/local.py [hunk 3]
```diff
@@ -1,45 +1,71 @@
...
+        recipe = connector.get(Reference.from_str(f"ReportRecipe|{report_task.report_recipe_id}"), valid_time)
+        report_types = [get_report_by_id(report_type_id) for report_type_id in recipe.report_types]
 
         error_reports, report_data = collect_reports(
-            now, connector, list(recipe.input_recipe["input_oois"]), parsed_report_types
+            valid_time, connector, recipe.input_recipe["input_oois"], report_types
         )
 
+        self.katalogus_client.organization = report_task.organisation_id
+        self.katalogus_client.organization_uri = f"/v1/organisations/{report_task.organisation_id}"
```
[the remark this answers is not in the document]
Comment (ammar92) [reply rocky/reports/runner/local.py:37]: The same goes for the bytes client I think

**Review by ammar92 — APPROVED** [review]
Looks good to me. I left a tiny suggestion for the clients for a next iteration

File: rocky/reports/runner/local.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (underdarknl) [reply rocky/reports/runner/local.py:37]: The client should (i think) be instantiated For a given organization, and possibly from the Organization Object. If you can instantiate the organization, you have no business connecting to its subsequent related api's.

[the remark this answers is not in the document]
Comment (Donnype) [reply rocky/reports/runner/local.py:37]: Some APIs of those clients are not organization-specific though, and hence forcing an organization as a property of the API client becomes a bit ugly. I believe this client is not the place to force handing over the organization as that makes the clients do to much and make them not reusable in context where we should have full access, such as with the runner. To have a nice layered design, we could introduce a layer where we properly manage access, such as a new client class that can only call organization specific methods of the base client.

**Comment by Donnype** [comment]
@Rieven Could you test this one again?

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Requires one small addition as described below. 

### What doesn't work:
By changing the format for the "Report name format" (first textbox) the corresponding explanation text is now outdated. This should be updated to explain the usage of the {oois_count}. 

image

### Bug or feature?:
n/a

**Comment by Donnype** [comment]
Thanks @stephanie0x00! Done

**Comment by stephanie0x00** [comment]
Looks good :+1:

File: rocky/katalogus/client.py [hunk 4]
```diff
@@ -129,7 +129,7 @@ def __init__(self, error: httpx.HTTPStatusError):
 
 
 class KATalogusClientV1:
-    def __init__(self, base_uri: str, organization: str):
+    def __init__(self, base_uri: str, organization: str | None):
         self.session = httpx.Client(base_url=base_uri)
         self.organization = valid_organization_code(organization) if organization else organization
         self.organization_uri = f"/v1/organisations/{organization}"
```
Comment (underdarknl) [inline rocky/katalogus/client.py:135]: this url would be malformed if no orga is present

Comment (Donnype) [reply rocky/katalogus/client.py:135]: Also see https://github.com/minvws/nl-kat-coordination/issues/3697, which I will pick up right away

**Review by underdarknl — APPROVED** [review]

**Pull request merged**